### DISCUSSION TITLE
Products variants with user input?

### DISCUSSION THREAD

**Discussion by Jakob-em** [saleor/saleor#8930] [Q&A] [answered]
Hi and thanks for this great project!

I was wondering if saleor supports products or product variants with user input out of the box?
For example a t shirt which has a customizable text which the user can enter before putting it in his cart.

I saw a few closed issues that address this topic but it isn't clear to me if this was implemented or not.

**Comment by karolkielecki** [comment]
@Jakob-em Hi! Thanks for your kind words! Re your question -- it's not supported, at least for now. I'll discuss it with the team and keep you updated.

**Comment by timuric** [comment, marked as answer]
Hi @Jakob-em currently you could store such info in the checkout metadata and later transfer it to the metadata of the order. The tricky bit is that at the time of the order creation checkout would be deleted, so you would have to tackle that. 

Alternatively you can extend orderLine class to support metadata, should be fairly straightforward, use Order model as a reference https://github.com/saleor/saleor/blob/main/saleor/order/models.py#L420
